wordpress-mobile/WordPress-FluxC-Android
# [Nullability Annotations to Java Classes] Enable and Configure `UnknownNullness` Android Lint Check

Parent #2795
Closes #2822

This PR enabled and configures the extra interop related `UnknownNullness` Android Lint check to help detect and flag some of the nullability related interoperability issues, that is, issues in Java for Kotlin consumption.

For more info see:

- [Docs](https://developer.android.com/kotlin/interop#supported_checks)
- [Check](https://googlesamples.github.io/android-custom-lint-rules/checks/UnknownNullness.md.html)

-----

## To test:
- There is nothing much to test here, verifying that all the CI checks are successful should be enough
    - Focus more on the Lint related ones.
- You could also check that the `Interoperability:Kotlin Interoperability` related section on the Lint report is shown and reporting some few thousand of warnings ([fluxc](https://buildkiteartifacts.com/b26f1746-2f0c-49cc-bbd1-a45a5e24c351/32e7e20c-4c4c-43b8-a886-f43cec935749/018a4ba5-253e-420a-8265-f8f81c4add54/018a4ba5-4c56-404a-a468-4c0123e2ce75/fluxc/build/reports/lint-results-release.html?response-content-type=text%2Fhtml&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQPCP3C7L7VCESSXK%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T130306Z&X-Amz-Expires=600&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEAgaCXVzLWVhc3QtMSJHMEUCIQCmX96wlQqMoEWhI554ixIbOeRz5HozCC9GYJUJ35EVcwIgX8EITaYXGt3Y9YUo5gyVaHNAy%2BaqYg0RWwpoD3ADj7oq%2BgMI0f%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwwMzIzNzk3MDUzMDMiDESiXBx%2BZHbnYg%2FEOyrOA2EeMZB7gq%2BXs6ITriodGigpeoiu8voNImcYRsp7yGWddyy99FxhEFbb698GSXV2R4Gkm07LKSQFQbvO6Sc6SJSq16AfanJNqtJezL50mYJ58UNeVY4s9ClntH22Y2iVhEHwzzCJQ7uR9TPhHktzvG4eKuLtUBWfoDMd3kRWBmfF44y7boqdpi1rhBJjhdWwN47foG1aNfunaPZXYXhZYNr3V41qt%2BFdOzaiRKAFXwr6vYyR4ICeV18kSluvngcF7G%2BiQ6xktnhqHyTACcbtXRQKet6cxaElIuiWmwNFGpRgfVssJmL72VQ41g1APuiBm3B5OxERvTni6FPDshlAVhU7nRhuLKuBbpgK%2Byg%2Faa7nfnOvIHJdDtRNC7G5A1wEIaW7fDyUEcopCtnjqqnjGTEOcWyHNet2fuxyCg9q50IrTMVyODKJuy0nDU%2B8MzNdCZr7VQbVN3rn8WrF%2B7CJmcVw5M3OArMwgfZz%2BbdnqhF0jH8R5WS40WsJ0cqOkNRnuxNf50tr9B39%2F5tJcN1miNb0XjFKKfuPLx9nIcz0ZAsMchtd16EjAXizGNPVP0T1Cv9Ihn3eLtFw8A2ggLtzgjatVMi8rrf3f%2FbsoiN8GjDygcGnBjqlAWz1QIl4qC6isHqgW1JStbrT4CIsgAY28OoTRygpDHwhVgO7MysUF0782ps%2BJdh8EuEUoZb30Dofj%2FYD2wuw29ADF6qSnkEDYPUrAuxyUgiKaWTael20RPoQhx2SsljXYVKekdIHSCfPL8dA9818CFQAljp%2FkLpo%2FFrHIdA%2F90K8Bfy0usD%2BKRwyyoOSwMwS3U86Wf5nGuax3K5n4k8tGgKwu2KqWw%3D%3D&X-Amz-SignedHeaders=host&X-Amz-Signature=fae5092781e3882efa6f5b2fa656c9abbb3187eeff1dca81250de2e80c86cc6d) + [plugins:woocommerce](https://buildkiteartifacts.com/b26f1746-2f0c-49cc-bbd1-a45a5e24c351/32e7e20c-4c4c-43b8-a886-f43cec935749/018a4ba5-253e-420a-8265-f8f81c4add54/018a4ba5-4c56-404a-a468-4c0123e2ce75/plugins/woocommerce/build/reports/lint-results-release.html?response-content-type=text%2Fhtml&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQPCP3C7LU5SKBKBQ%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T130308Z&X-Amz-Expires=600&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEAgaCXVzLWVhc3QtMSJHMEUCIDGhtR1jgmE3sJoHEYHp9Ch3M%2BoTd49TyDRuCWqUwz%2BwAiEAs%2BkAM0WS3%2BztyxCbPw9dlOv7vlpKhDwLVCAjo8LuE8oq%2BgMI0f%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwwMzIzNzk3MDUzMDMiDPJIjn9oLLDLCM1fsirOA%2BV2OZD21g%2FEE1WS0lCLZvy2LqHqiIarQTwt%2BC3agzw6tURAM0eMOsG%2Fc1XnS4b0YPUGJxzmSXaqLy%2FcmKPT6zrHe4sX7uIrURqn6cq7Zgzz23hg5vCDlKUYBTpu4Dh3opKhih4BpKwAxgbhE2pl6Xa1XbUtHeZvyse2eyN0uFR6zJsLaDwPX%2FPr691A9MPzx7dWOwN2swdCL5bz%2Fz0ZIBw5UdXate7JMaoCf8pPuwXkrjIqx5KLjW4pBWlSI7BtuQb8Y%2FGk8dQ2gJ%2BZ5bcBf65HQplfYDLHYf8ceqvvpBCPINe2Nll9WiQGhFbAnzB%2BgEWoyP5rT4Wm6pywmBl9XzWp2CGwf0RE%2Bw48Ok3XzRFe3Lgnim9JDAp6yI0N5jMwpL0FKDNl2YSp4WjZT9amHFmpqiUkVUKO2ixGQMS0JD0uxJXx6N%2BRN992Somj80SzpEpFA%2BgxJtBW9%2FVz8SNt36clWhmTfSRwqhtwGkLPTUlEUvZua9OGhZaXvUKdRyO1PKbNOvEMev6Rkn4Wv9g2DiZ4uQ%2FHQIq6jLhsXgYqWyOzXBYgWTDQFlL8LtunTgvtoHa9kEsIckIx%2FoxdniwbzWFH7hstiIlN0uUfcj%2BZUzDbgMGnBjqlAeDl5S%2BGPa%2B7JDrMLuS8LArwwAHaVsO83r5DhxM8z%2BvWsSkEzOwTDHil0INWV7QlRoMVFFcFhdDr%2F%2B%2BBlR7DO8gP%2FMGLqVCEE3AxLk6jYqiDFSD84auE1BcGuziSvdQLlSnjdpDGppUyK9gS29j0W9jGxg0sk%2BnBn61L6mLGBRD8PyzsOr2OBc9h23XCOOTvCp1K2jpv8R4nUtcnC1mDeLFqNqlvcg%3D%3D&X-Amz-SignedHeaders=host&X-Amz-Signature=f763d0523620a988816ae1242a410c29070ebca7a8bb4e66eb406bcea5ab8fe9) + [example](https://buildkiteartifacts.com/b26f1746-2f0c-49cc-bbd1-a45a5e24c351/32e7e20c-4c4c-43b8-a886-f43cec935749/018a4ba5-253e-420a-8265-f8f81c4add54/018a4ba5-4c56-404a-a468-4c0123e2ce75/example/build/reports/lint-results-release.html?response-content-type=text%2Fhtml&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQPCP3C7LQBAYSM45%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T130312Z&X-Amz-Expires=600&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEAgaCXVzLWVhc3QtMSJHMEUCID4rHb8LW0cTtHvt5im2slv8nvFrzVXREgQ3svOnwCjYAiEAs3M7i3iOMa675JNxW6ZQgZQ%2FAIv3DMPbg4ePplA13sIq%2BgMI0f%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwwMzIzNzk3MDUzMDMiDAy%2B%2BoIX7rhAaqhzoSrOA6khp8rXU6STI81Wbt5YH3FFHjDJ3BdiNkDKUH4HAiYgoay14uB5J8wdvNgeVOSj58LeXwcDNvYyTXRF0mnHOTKgNTrnwSORJzQekQEBEQehsPVG%2B3gz9enafC%2FLkvon%2Fae%2BisuOe60gEtxX7hoZ9uhRFjE%2FarNxjR0i4K0wDvo74fvchM0IiG9z0qGc59idyvOyg%2F%2FIfGsVt1bpk02J5JJauATgQNw3rtWZCu54rSdCA2V1Qq7pA13MnAngmkUdpuY%2FWejwy2bhkso0k6rSwUnDOcJLpElS7lSxju%2BQwdkdf47ynGRzh6pYY6e%2BkytRBXsMsRkmkGFW1RQVd3Sh%2FvgMbQ8fNvQVNffToXhOmhRkU6D3%2FCL%2BaQUA%2BIRF9NmVGIIRFjzg6i8%2B4idQTz0r0nuXyAAep44wM64p1HKQIclvlhIocYG1vh7NiioUyyVPBkrUcH82qqB0gFh%2BMXjCALb4qryKEEqM88ySObx24Kjs6xr%2BeJgWWoCG%2FJgsGUij89ZHSMyIAqW%2F%2FJvI5bmTYbjkalGQt%2FsUr96Q%2BItXe6PvjEvqbDGxhrN%2FIY8Vjm374Afy3ncSZrzFZtUbwB6iNIa%2BD7zHpWkuk0hnL9NVrTCl%2F8CnBjqlAb31YzexWC88uMYmScqXi70Eg8HqnB2RbPeHdM9quSp%2BK6Mrg6jNOv8MqwaJ%2FBZMeA10hPmv48xrnssiHBN0WmHGgHM1lbP5y34KF1CJXpEvBC5MYzv11nEg%2ByjCtJLgrn0p4RxieR%2BGu1JeA%2FGTNyRRniCEd78vKbtTay%2BZ33IUyjIJKVEziaGHSP6x1NqKTBnhEUsu%2BTdUbJSAaYVlM%2BZbPbhmHQ%3D%3D&X-Amz-SignedHeaders=host&X-Amz-Signature=769b81dcc704f15b86a397e6efb4e17672b68a1b5c4fb04e3c83d450e1a1422e)).
    - For example, locally, running Lint on the `fluxc` + `debug` variant, I got `2152` such warnings, configured to be informational only so that they don't fail the build, see screenshot below:

![image](https://github.com/wordpress-mobile/WordPress-FluxC-Android/assets/9729923/8f4d37dc-6ce9-43c4-98a7-1d1e84a0bc4c)
